v3/pforzheim.py

```diff
@@ -0,0 +1,113 @@
+"""
+Copyright 2024 binary butterfly GmbH
+Use of this source code is governed by an MIT-style license that can be found in the LICENSE.txt.
+"""
+
+import csv
+import json
+from datetime import datetime, timezone
+from decimal import Decimal
+from io import StringIO
+from typing import Optional
+
+from validataclass.dataclasses import validataclass
+from validataclass.exceptions import ValidationError
+from validataclass.validators import BooleanValidator, DataclassValidator, DecimalValidator, IntegerValidator, StringValidator
+
+from common.base_converter import JsonConverter
+from common.exceptions import ImportParkingSiteException
+from common.models import ImportSourceResult
+from common.validators import StaticParkingSiteInput
+from common.validators.base_validators import ParkingSiteTypeInput
+from common.validators.fields.noneable import ExcelNoneable
+from util import SourceInfo, name_to_id
+
+
+@validataclass
+class PforzheimRowInput:
+    uid: str = StringValidator(max_length=255)
+    name: str = StringValidator(max_length=255)
+    operator_name: str = StringValidator(max_length=255)
+    address: str = StringValidator(max_length=255, multiline=True)
+    description: str = StringValidator(max_length=512, multiline=True)
+    type: str = StringValidator(max_length=255)
+    lat: Decimal = DecimalValidator(min_value=40, max_value=60)
+    lon: Decimal = DecimalValidator(min_value=7, max_value=10)
+    # To Do, the capacity attributes are initialized in the case of empty strings e.g ''
+    capacity: Optional[int] = ExcelNoneable(IntegerValidator(allow_strings=True), default=1)
+    capacity_woman: Optional[int] = ExcelNoneable(IntegerValidator(allow_strings=True), default=0)
+    capacity_disabled: Optional[int] = ExcelNoneable(IntegerValidator(allow_strings=True), default=0)
+    is_supervised: bool = BooleanValidator()
+    fee_description: str = StringValidator(multiline=True)
+    opening_hours: str = StringValidator(max_length=255, multiline=True)
+
+
+class PforzheimConverter(JsonConverter):
+    pforzheim_row_validator = DataclassValidator(PforzheimRowInput)
+
+    source_info = SourceInfo(
+        id='pforzheim',
+        name='Stadt Pforzheim',
+        public_url='https://www.pforzheim.de',
+    )
+
+    type_mapping: dict[str, ParkingSiteTypeInput] = {
+        'onStreet': ParkingSiteTypeInput.ON_STREET,
+        'carPark': ParkingSiteTypeInput.CAR_PARK,
+        'undergroundCarPark': ParkingSiteTypeInput.UNDERGROUND,
+    }
+
+    def handle_json(self, data: dict | list) -> ImportSourceResult:
+        import_source_result = self.generate_import_source_result(
+            static_parking_site_inputs=[],
+            static_parking_site_errors=[],
+        )
+
+        # We start at row 2, as the first one is our header
+        for item in data:
+            input_dict = {
+                'uid': item.get('Id'),
```

Comment:
As `validataclass`  is optimized for JSON input, you don't need to do a remapping by `get` at all. Just write the `@validataclass` like this:

```
@validataclass
class PforzheimInput:
    Id: str = StringValidator()
    quantitySpacesReservedForMobilityImpededPerson: Optional[int] = Noneable(IntegerValidator)
    ...
```

And map the `validataclass result to `StaticParkingSiteInput` fields afterwards:

```
StaticParkingSiteInput(
    uid=input_data.Id,
    capacity_disabled=input_data.quantitySpacesReservedForMobilityImpededPerson
    ...
```